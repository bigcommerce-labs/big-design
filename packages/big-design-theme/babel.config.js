module.exports = require('@bigcommerce/configs/babel');
