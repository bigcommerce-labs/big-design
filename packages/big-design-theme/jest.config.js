const defaultJestConfig = require('@bigcommerce/configs/jest');

module.exports = {
  ...defaultJestConfig,
  coverageThreshold: {
    global: {
      statements: 100,
      branches: 100,
      functions: 100,
      lines: 100,
    },
  },
};
