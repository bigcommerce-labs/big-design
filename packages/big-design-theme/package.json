{
  "name": "@bigcommerce/big-design-theme",
  "version": "0.12.1",
  "sideEffects": false,
  "main": "dist/cjs/index.js",
  "module": "dist/es/index.js",
  "typings": "dist/index.d.ts",
  "license": "SEE LICENSE IN LICENSE.md",
  "author": "BigCommerce Inc",
  "repository": {
    "type": "git",
    "url": "https://github.com/bigcommerce/big-design.git",
    "directory": "packages/big-design-theme"
  },
  "scripts": {
    "build": "yarn run build:cjs && yarn run build:es && yarn run build:dt",
    "build:cjs": "NODE_ENV=production BABEL_ENV=cjs babel --extensions \".ts,.tsx\" ./src --out-dir ./dist/cjs",
    "build:es": "NODE_ENV=production BABEL_ENV=es babel --extensions \".ts,.tsx\" ./src --out-dir ./dist/es",
    "build:dt": "tsc --emitDeclarationOnly",
    "ci": "yarn run lint && yarn run test --maxWorkers=2 --coverage && yarn run build",
    "lint": "yarn run lint:tsc",
    "lint:tsc": "tsc --noEmit",
    "prepublishOnly": "rimraf dist && yarn run ci",
    "test": "BABEL_ENV=test jest",
    "test:watch": "yarn run test --watch"
  },
  "files": [
    "dist"
  ],
  "publishConfig": {
    "access": "public"
  },
  "dependencies": {
    "@babel/runtime": "^7.12.1",
    "polished": "^3.0.3"
  },
  "peerDependencies": {
    "react": "^16.8.0 || ^17.0.0",
    "react-dom": "^16.8.0 || ^17.0.0",
    "styled-components": "^4.3.0"
  },
  "devDependencies": {
    "@babel/cli": "^7.12.1",
    "@babel/core": "^7.12.3",
    "@babel/plugin-proposal-class-properties": "^7.12.1",
    "@babel/plugin-proposal-nullish-coalescing-operator": "^7.12.1",
    "@babel/plugin-proposal-object-rest-spread": "^7.12.1",
    "@babel/plugin-proposal-optional-chaining": "^7.12.1",
    "@babel/plugin-transform-object-assign": "^7.12.1",
    "@babel/plugin-transform-runtime": "^7.12.1",
    "@babel/preset-env": "^7.12.1",
    "@babel/preset-react": "^7.12.1",
    "@babel/preset-typescript": "^7.12.1",
    "@bigcommerce/configs": "^0.14.1",
    "@types/jest": "^25.2.1",
    "@types/styled-components": "^4.1.12",
    "babel-jest": "^25.5.1",
    "babel-plugin-styled-components": "^1.10.6",
    "concurrently": "^5.0.0",
    "jest": "^25.5.4",
    "jest-styled-components": "^6.3.1",
    "react": "^17.0.1",
    "react-dom": "^17.0.1",
    "rimraf": "^3.0.2",
    "styled-components": "^4.3.0",
    "typescript": "~3.8.3",
    "typescript-styled-plugin": "^0.15.0"
  }
}
