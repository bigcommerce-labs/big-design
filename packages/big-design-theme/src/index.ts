import { themeOptions, ThemeOptions } from './options';
import { createBorder, createBorderRadius, Border, BorderRadius } from './system/border';
import { breakpoints, breakpointValues, Breakpoints, BreakpointValues } from './system/breakpoints';
import { colors, Colors } from './system/colors';
import * as keyframes from './system/keyframes';
import { createLineHeight, LineHeight } from './system/line-height';
import { shadow, Shadow } from './system/shadow';
import { createSpacing, Spacing } from './system/spacing';
import { createTypography, Typography } from './system/typography';
import { zIndex, ZIndex } from './system/z-index';

export * from './helpers';
export * from './system';

export interface ThemeInterface {
  border: Border;
  borderRadius: BorderRadius;
  breakpointValues: BreakpointValues;
  breakpoints: Breakpoints;
  colors: Colors;
  keyframes: typeof keyframes;
  lineHeight: LineHeight;
  shadow: Shadow;
  spacing: Spacing;
  typography: Typography;
  zIndex: ZIndex;
}

export const createTheme = (customOptions: Partial<ThemeOptions> = {}): ThemeInterface => {
  themeOptions.setOptions(customOptions);

  return {
    border: createBorder(),
    borderRadius: createBorderRadius(),
    breakpointValues,
    breakpoints,
    colors,
    keyframes,
    lineHeight: createLineHeight(),
    shadow,
    spacing: createSpacing(),
    typography: createTypography(),
    zIndex,
  };
};

export const theme: ThemeInterface = createTheme();
