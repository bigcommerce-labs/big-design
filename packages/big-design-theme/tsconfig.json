{
  "extends": "@bigcommerce/configs/ts",
  "compilerOptions": {
    "declaration": true,
    "declarationDir": "dist",
    "declarationMap": true,
    "jsx": "preserve",
    "module": "esnext",
    "target": "esnext"
  },
  "include": ["src"],
  "exclude": ["dist"],
}
