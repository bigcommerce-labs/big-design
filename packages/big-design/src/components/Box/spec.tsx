import { theme } from '@bigcommerce/big-design-theme';
import { render, screen } from '@testing-library/react';
import React, { createRef } from 'react';
import 'jest-styled-components';

import { Box } from './index';

test('has margin props', () => {
  const { rerender } = render(<Box>test</Box>);

  expect(screen.getByText('test')).not.toHaveStyle('margin: 1rem');

  rerender(<Box margin="medium">test</Box>);

  expect(screen.getByText('test')).toHaveStyle('margin: 1rem');
});

test('has padding props', () => {
  const { rerender } = render(<Box>test</Box>);

  expect(screen.getByText('test')).not.toHaveStyle('padding: 1rem');

  rerender(<Box padding="medium">test</Box>);

  expect(screen.getByText('test')).toHaveStyle('padding: 1rem');
});

test('has backgroundColor props', () => {
  const { rerender } = render(<Box>test</Box>);

  expect(screen.getByText('test')).not.toHaveStyle('background-color: #fff');

  rerender(<Box backgroundColor="white">test</Box>);

  expect(screen.getByText('test')).toHaveStyle('background-color: #fff');
});

test('has borderRadius props', () => {
  const { rerender } = render(<Box>test</Box>);

  expect(screen.getByText('test')).not.toHaveStyle('border-radius: 50%');

  rerender(<Box borderRadius="circle">test</Box>);

  expect(screen.getByText('test')).toHaveStyle('border-radius: 50%');
});

test('has border props', () => {
  render(<Box border="box">test</Box>);

  expect(screen.getByText('test')).toMatchSnapshot();
});

test('has individual border props', () => {
  render(
    <Box borderBottom="box" borderLeft="box" borderRight="box">
      test
    </Box>,
  );

  expect(screen.getByText('test')).toMatchSnapshot();
});

test('has shadow props', () => {
  render(<Box shadow="floating">test</Box>);

  expect(screen.getByText('test')).toMatchSnapshot();
});

test('has clearfix prop', () => {
  render(<Box clearfix>test</Box>);

  expect(screen.getByText('test')).toMatchSnapshot();
});

test('has zIndex props', () => {
  const { rerender } = render(<Box zIndex="sticky">test</Box>);

  expect(screen.getByText('test')).toHaveStyle(`z-index: ${theme.zIndex.sticky}`);

  rerender(<Box zIndex="popover">test</Box>);

  expect(screen.getByText('test')).toHaveStyle(`z-index: ${theme.zIndex.popover}`);
});

test('renders as a different tag', () => {
  render(<Box as="section">test</Box>);

  expect(screen.getByText('test').tagName).toBe('SECTION');
});

test('box forwards ref', () => {
  const ref = createRef<HTMLDivElement>();

  render(<Box ref={ref}>test</Box>);

  expect(screen.getByText('test')).toBe(ref.current);
});
