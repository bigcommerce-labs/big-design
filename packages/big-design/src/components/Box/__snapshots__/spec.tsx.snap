// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`has border props 1`] = `
.c0 {
  box-sizing: border-box;
  border: 1px solid #D9DCE9;
}

<div
  class="c0"
/>
`;

exports[`has individual border props 1`] = `
.c0 {
  box-sizing: border-box;
  border-right: 1px solid #D9DCE9;
  border-bottom: 1px solid #D9DCE9;
}

<div
  class="c0"
/>
`;

exports[`has shadow props 1`] = `
.c0 {
  box-sizing: border-box;
  border-radius: 0.25rem;
  box-shadow: 0px 2px 12px rgba(49,52,64,0.2);
}

<div
  class="c0"
/>
`;
